src/api/rasters.ts

```diff
@@ -0,0 +1,290 @@
+  const response = await fetch(
+    // show all temporal rasters the user has access to
+    `/api/v3/rasters/?page_size=0&name__icontains=${searchString}&first_value_timestamp__isnull=false`,
+    {
+      credentials: "same-origin"
+    }
+  );
+
+  return response.json();
+};
+
+
+export const listRastersForTable = async (
+  pageSize: number, page: number, searchString: string, organisationUuid: string
+) => {
+  const url = `/api/v4/rasters/?writable=true&page_size=${pageSize}&page=${page}&name__icontains=${searchString}&ordering=last_modified&organisation__uuid=${organisationUuid}&scenario__isnull=true`;
+
+  const response = await fetch(url, {
+    credentials: "same-origin"
+  });
+
+  return response.json();
+};
+
+export const uploadRasterFile = async (rasterUuid: string, file: File, timestamp: Date) => {
+  const form = new FormData();
+
+  form.append("file", file);
+
+  if (timestamp !== undefined) {
+    form.append("timestamp", timestamp.toISOString());
+  }
+
+  const url = `/api/v4/rasters/${rasterUuid}/data/`;
+
+  const opts: RequestInit = {
+    credentials: "same-origin",
+    method: "POST",
+    headers: {
+      mimeType: "multipart/form-data"
+    },
+    body: form
+  };
+
+  const response = await fetch(url, opts);
+  return response.json();
+};
+
+
+export const createRaster = async (raster: OldRaster) => {
+  // A raster is created in two steps: first we create the raster *source*,
+  // then the raster *layer* using the uuid of the first.
+  // Some of the fields go into the source, some into the layer, some into both.
+
+  const defaultOpts: RequestInit = {
+    credentials: "same-origin",
+    method: "POST",
+    headers: { "Content-Type": "application/json" },
+  };
+
+  const rasterSourceBody = {
+    name: raster.name,
+    description: raster.description,
+    organisation: raster.organisation,
+    access_modifier: "Private", // Always
+    supplier: raster.supplier,
+    supplier_code: raster.supplier_code,
+    temporal: raster.temporal,
+    interval: raster.temporal ? raster.interval : undefined
+  };
+
+  const rasterSourceResponse = await fetch('/api/v4/rastersources/', {
+    ...defaultOpts,
+    body: JSON.stringify(rasterSourceBody)
+  });
+
+  if (rasterSourceResponse.status !== 201) {
+    // Status other than "Created"
+    return rasterSourceResponse; // Don't know what else to do
+  }
+
+  const rasterSource = await rasterSourceResponse.json();
+
+  const rasterBody = {
+    name: raster.name,
+    organisation: raster.organisation,
+    observation_type: raster.observation_type,
+    description: raster.description,
+    supplier: raster.supplier,
+    aggregation_type: raster.aggregation_type,
+    options: raster.options,
+    shared_with: raster.shared_with,
+    rescalable: raster.rescalable,
+    access_modifier: raster.access_modifier,
+    temporal: raster.temporal,
+    interval: raster.temporal ? raster.interval : undefined,
+    source: {
```

Comment:
It does feel strange to have to refer to 'lizard_nxt.blocks.LizardRasterSource' in a client. I don't know about the rest of the structure, is it going to be used for other things?

Since we look at the raster_sources property when getting a raster, it could also make sense to POST to raster_sources: [<url of raster source>] ?

Reply:
> It does feel strange to have to refer to 'lizard_nxt.blocks.LizardRasterSource' in a client. I don't know about the rest of the structure, is it going to be used for other things?

The whole geoblocks story is in there; the "source" field accepts a graph with any number of keys ("blocks"). The one that is now hardcoded in this PR is a special case of that. Adding the possibility of directly writing a UUID would be adding "another way to do it". As we are close the release and there are many other bugs to fix, I tend to leave it as is.

> Since we look at the raster_sources property when getting a raster, it could also make sense to POST to raster_sources: [] ?

A POST to raster_sources would be complicated to build. It is meant as a read-only field to easily list the sources that are in the `"source"` field.

Reply:
Leaving that as is then.